Update @woo.com email addresses to @woocommerce.com

lsinger opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Updates all Woo.com references in email addresses to WooCommerce.com, a complementary PR to #46259. 

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. ensuring the change does not break any tests
2. ensure through manual inspection that the email addresses look correct (note: receiving has been confirmed to work for both domains)

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

lsinger committed:
update @woo.com email addresses

lsinger committed:
update %40woo.com email address

lsinger committed:
add changelog file

lsinger committed:
update docs manifest

coreymckrill reviewed (commented):
@lsinger looks good, although when I did I search through the codebase using PhpStorm after checking out this branch and running `pnpm build`, I found a couple more instances of `@woo.com`:

```
* plugins/woocommerce/packages/woocommerce-blocks/assets/js/blocks/classic-template/order-confirmation.tsx:120
* plugins/woocommerce/packages/woocommerce-blocks/build/legacy-template.js:7
```

There also might be a couple more if you rebase this to trunk (sorry for the delay in reviewing this)

lsinger commented:
> @lsinger looks good, although when I did I search through the codebase using PhpStorm after checking out this branch and running `pnpm build`, I found a couple more instances of `@woo.com`:
> 
> ```
> * plugins/woocommerce/packages/woocommerce-blocks/assets/js/blocks/classic-template/order-confirmation.tsx:120
> * plugins/woocommerce/packages/woocommerce-blocks/build/legacy-template.js:7
> ```
> 
> There also might be a couple more if you rebase this to trunk (sorry for the delay in reviewing this)

Thanks @coreymckrill! I resolved merge conflicts and rebased. 

I still found the files you mentioned, but realized that those are all in build or cache folders -- and as such are all in .gitignore (either at the root repository level or in a project directory). I ran `pnpm clean` from the repo root and then `egrep -lir "(@woo.com)" .`, which yielded a few files in `build` and `assets` directories. 

They're both associated with builds and ignored, so I deleted them locally, which made no difference to git. FWIW, re-building also didn't recreate them (and they're curiously not covered by any `clean` tasks), so my current suspicion is that these were build artifacts from past build processes that may have changed over time. 

With that background information I'll go ahead and request another review from you -- let me know what you think! ☺️

coreymckrill committed:
Update docs manifest

coreymckrill reviewed (approved):
👍 Looks good. There was a merge conflict so I rebased to trunk and updated the hash in the docs manifest again
